### DISCUSSION TITLE
[New Component] [Message Box] can i add one more button ? and can i disable  how to When i click anywhere outside the messagebox, disable close

### DISCUSSION THREAD

**Discussion by Meguais** [element-plus/element-plus#11571] [Q&A]
Feature type: **`New Component`**

## Details

- New component name: `Message Box`

## Description

can i add one more button ? and can i disable 
how to When i click anywhere outside the messagebox, disable close
https://cdn.discordapp.com/attachments/902897406394392576/1074814685687320606/image.png

### Additional comments

_(empty)_

**Comment by btea** [comment]
>how to When i click anywhere outside the messagebox, disable close    

You can set `close-on-click-modal` value to `false`.

> can i add one more button ? and can i disable

You can use vnode to customize the implementation, [example](https://element-plus.run/#eyJBcHAudnVlIjoiPHRlbXBsYXRlPlxuICA8ZWwtYnV0dG9uIHBsYWluIEBjbGljaz1cIm9wZW5cIj5Db21tb24gVk5vZGU8L2VsLWJ1dHRvbj5cbjwvdGVtcGxhdGU+XG5cbjxzY3JpcHQgbGFuZz1cInRzXCIgc2V0dXA+XG5pbXBvcnQgeyBoLCByZWYgfSBmcm9tICd2dWUnXG5pbXBvcnQgeyBFbE1lc3NhZ2VCb3gsIEVsU3dpdGNoLCBFbEJ1dHRvbiB9IGZyb20gJ2VsZW1lbnQtcGx1cydcbmNvbnN0IGNhbmNlbCA9ICgpID0+IHtcbiAgRWxNZXNzYWdlQm94LmNsb3NlKClcbn1cbmNvbnN0IG9wZW4gPSAoKSA9PiB7XG4gIEVsTWVzc2FnZUJveCh7XG4gICAgdGl0bGU6ICdNZXNzYWdlJyxcbiAgICBzaG93Q2FuY2VsQnV0dG9uOiBmYWxzZSxcbiAgICBzaG93Q29uZmlybUJ1dHRvbjogZmFsc2UsXG4gICAgY2xvc2VPbkNsaWNrTW9kYWw6IGZhbHNlLFxuICAgIG1lc3NhZ2U6IGgoJ3AnLCB7IHN0eWxlOiAndGV4dC1hbGlnbjogY2VudGVyOycgfSwgW1xuICAgICAgaChFbEJ1dHRvbiwgeyB0eXBlOiAncHJpbWFyeScsIGlubmVyVGV4dDogJ2NhbmNlbCcsIG9uY2xpY2s6IGNhbmNlbCB9KSxcbiAgICAgIGgoRWxCdXR0b24sIHsgdHlwZTogJ3ByaW1hcnknLCBpbm5lclRleHQ6ICdjb25maXJtJywgb25jbGljazogY2FuY2VsIH0pLFxuICAgICAgaChFbEJ1dHRvbiwgeyB0eXBlOiAnZGFuZ2VyJywgaW5uZXJUZXh0OiAnZXhpdCcsIG9uY2xpY2s6IGNhbmNlbH0pXG4gICAgXSksXG4gIH0pXG59XG5cbjwvc2NyaXB0PlxuIiwiaW1wb3J0X21hcC5qc29uIjoie1xuICBcImltcG9ydHNcIjoge31cbn0iLCJfbyI6e319).

**Comment by Meguais** [comment]
thank you can i change custom background color ?

**Reply by btea** [reply]
like [this](https://element-plus.run/#eyJBcHAudnVlIjoiPHRlbXBsYXRlPlxuICA8ZWwtYnV0dG9uIHBsYWluIEBjbGljaz1cIm9wZW5cIj5Db21tb24gVk5vZGU8L2VsLWJ1dHRvbj5cbjwvdGVtcGxhdGU+XG5cbjxzY3JpcHQgbGFuZz1cInRzXCIgc2V0dXA+XG5pbXBvcnQgeyBoLCByZWYgfSBmcm9tICd2dWUnXG5pbXBvcnQgeyBFbE1lc3NhZ2VCb3gsIEVsU3dpdGNoLCBFbEJ1dHRvbiB9IGZyb20gJ2VsZW1lbnQtcGx1cydcbmNvbnN0IGNhbmNlbCA9ICgpID0+IHtcbiAgRWxNZXNzYWdlQm94LmNsb3NlKClcbn1cbmNvbnN0IG9wZW4gPSAoKSA9PiB7XG4gIEVsTWVzc2FnZUJveCh7XG4gICAgdGl0bGU6ICdNZXNzYWdlJyxcbiAgICBzaG93Q2FuY2VsQnV0dG9uOiBmYWxzZSxcbiAgICBzaG93Q29uZmlybUJ1dHRvbjogZmFsc2UsXG4gICAgY2xvc2VPbkNsaWNrTW9kYWw6IGZhbHNlLFxuICAgIGN1c3RvbVN0eWxlOiB7XG4gICAgICBiYWNrZ3JvdW5kOiAnIzZjZidcbiAgICB9LFxuICAgIG1lc3NhZ2U6IGgoJ3AnLCB7IHN0eWxlOiAndGV4dC1hbGlnbjogY2VudGVyOycgfSwgW1xuICAgICAgaChFbEJ1dHRvbiwgeyB0eXBlOiAncHJpbWFyeScsIGlubmVyVGV4dDogJ2NhbmNlbCcsIG9uY2xpY2s6IGNhbmNlbCB9KSxcbiAgICAgIGgoRWxCdXR0b24sIHsgdHlwZTogJ3ByaW1hcnknLCBpbm5lclRleHQ6ICdjb25maXJtJywgb25jbGljazogY2FuY2VsIH0pLFxuICAgICAgaChFbEJ1dHRvbiwgeyB0eXBlOiAnZGFuZ2VyJywgaW5uZXJUZXh0OiAnZXhpdCcsIG9uY2xpY2s6IGNhbmNlbH0pXG4gICAgXSksXG4gIH0pXG59XG5cbjwvc2NyaXB0PlxuIiwiaW1wb3J0X21hcC5qc29uIjoie1xuICBcImltcG9ydHNcIjoge31cbn0iLCJfbyI6e319)?

**Comment by Meguais** [comment]
thank you 👯
